### DISCUSSION TITLE
Annotations support

### DISCUSSION THREAD

**Discussion by noirbizarre** [ludeknovy/jtl-reporter#231] [Ideas]
In most of my load/perf tests, I have some events triggered at some point and I would love being able to show them in the charts as an annotation (like. the [builtin ones](https://www.highcharts.com/docs/advanced-chart-features/annotations-module) or ideally custom vertical ones like [this](https://people.iola.dk/olau/flot/examples/annotating.html).
Another case I have where annotating is interesting is phased/staged load testing: when your run is divided into different phase. Ideally, you want to highlight the phases and vertical annotations on start would be great.

Is it a feature that someone is interested in ?

**Comment by ludeknovy** [comment]
Hi @noirbizarre ! That's an interesting idea. What tool do you use, or how would you provide timestamps for those annotations/ stages?

**Comment by noirbizarre** [comment]
It's currently in its very early shape so I don't have a full specification, but I wanted to make sure that such a feature would be welcome.

I am mostly using `locust` as a tool with a slightly modified `jtl-listener` from https://github.com/ludeknovy/jtl-reporter/blob/main/scripts/jtl_listener_service.py.

I am proposing to implement this on locust in parallel (cf. https://github.com/locustio/locust/issues/2397).

The ideal implementation would have both basic annotations (a label associated with a timestamp) and spans (a label associated with a time range) but annotations only at first is a good start.

In JtlReporter, after a quick look, I think it consists of:

- a backend implementation: 
  - a submission API (`POST /api/projects/<project>/scenarios/<scenario>/items/<id>/annotations`)
  - a listing submission API (`GET /api/projects/<project>/scenarios/<scenario>/items/<id>/annotations`)
  - a storage model
- a frontend implementation retrieving those annotations and using Highcharts API to display them

JtlReporter is kind of new to me, so I might be missings some details. Any insight is welcome so I can implement this (unless anyone is willing to take it)

**Reply by ludeknovy** [reply]
Thanks for the details.

I think the implementation would be slightly different on JtlReporer's side:
1) we would extend the current endpoint for CSV upload
2) create a new endpoint in jtl reporter service to it would support the same functionality but for continuous uploads.
3) returning the annotation data in report detail either from the new endpoint as you suggested or extend the existing one.

**Reply by ludeknovy** [reply]
@noirbizarre any progress on https://github.com/locustio/locust/issues/2397 ?
